src/react/databrowser/objects/details/ObjectRestorationButtonAndModal.tsx

```diff
@@ -0,0 +1,166 @@
+          </tr>
+        </thead>
+        <tbody>
+          <tr style={{ borderBottom: '1px solid #323245' }}>
+            <td align="left">{objectKey}</td>
+            <td align="left">
+              {formatShortDate(new Date(objectLastModifiesOn))}
+            </td>
+            <td align="left">
+              <PrettyBytes bytes={objectSize} decimals={1} />
+            </td>
+            <td align="left">{objectStorageClass || 'default'}</td>
+          </tr>
+        </tbody>
+      </table>
+    );
+  };
+  const RestorationModalContent = () => {
+    return (
+      <div style={{ maxWidth: '45rem' }}>
+        {restoreError && (
+          <Banner
+            variant="warning"
+            title="Restore Object Failed"
+            icon={<Icon name="Exclamation-circle" color="statusWarning"></Icon>}
+          >
+            {restoreError}
+          </Banner>
+        )}
+        <Stack direction="vertical" gap="f32">
+          <Text>
+            To restore an object, you need to launch a Restoration request, then
+            wait for the object to become available.
+          </Text>
+          <ObjectTable />
+          <Stack>
+            Duration of Restoration
+            <Input
+              id="restorationDuration"
+              name="restorationDuration"
+              type="number"
+              value={restorationDays}
+              onChange={(e: ChangeEvent<HTMLInputElement>) =>
+                setRestorationDays(parseInt(e.target.value))
+              }
+              size="1/3"
+            ></Input>{' '}
+            day(s)
+          </Stack>
+          <Text>
+            A temporary copy of your object will be available during a limited
+            number of days. After this period, the restored copy will be
+            automatically deleted.
+          </Text>
+        </Stack>
+      </div>
+    );
+  };
+
+  const restore = useMutation(
+    async () => {
+      await zenkoClient.restoreObject(
+        bucketName,
+        objectKey,
+        restorationDays,
+        objectVersionId,
+      );
+    },
+    {
+      onSuccess: () => {
+        handleClose();
+      },
+      onError: (err: AWSError) => {
+        setRestoreError(err.message);
+      },
+    },
+  );
+
+  return (
+    <>
+      <Button
+        id="restore-button"
+        type="button"
+        variant="outline"
+        disabled={isObjectRestoredOrOnGoing || restore.isLoading}
+        label="Restore"
+        onClick={() => setIsModalOpen(true)}
+      ></Button>
+      <Modal
+        close={handleClose}
+        footer={
+          <Wrap>
+            <p></p>
+            <Stack>
+              <Button variant="outline" onClick={handleClose} label="Cancel" />
+              <Button
+                variant="primary"
+                onClick={() => restore.mutate()}
+                label="Start Restoration"
+                disabled={restore.isLoading}
+              />
+            </Stack>
```

Comment:
nit: We may want to change the label or add a tooltip on top of the button to describe the disabled states when it is disabled